File: pharmacy_app/lib/core/Models/Medicine.dart
//import 'dart:ffi';

import 'Category.dart';
//import 'PharmacyMedicine.dart';

class Medicine {
  int? Id;
  String? Name;
  String? Description;
  String? ImageUrl;
  String? Company;
  String? MadeIn;
  double? Price;
  String? DateCreated;
  String? ExpirationDate;
  //List<PharmacyMedicine>? PharmacyMedicines;
  int? CategoryId;
  Category? category;

  Medicine({
    this.Id,
    this.Name,
    this.Description,
    this.ImageUrl,
    this.Company,
    this.MadeIn,
    this.Price,
    this.DateCreated,
    this.ExpirationDate,
    this.CategoryId,
    this.category,
    //this.PharmacyMedicines
  });

  factory Medicine.fromJson(Map<String,dynamic> json)
  {
    return Medicine(
      Id: json['id'] ?? 0,
      Name: json['name'] ?? '',
      Description: json['description'],
      ImageUrl: json['imageUrl'],
      Company: json['company'],
      MadeIn: json['madeIn'], 
      Price: json['price'],
      DateCreated: json['dateCreated'],
      ExpirationDate: json['expirationDate'],
      CategoryId: json['categoryId'],
      //category: Category.fromJson(json['category']),
    );
  } 
}

File: Pharmacy_System/Pharmacy_API/Models/Medicine.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Pharmacy_API.Models
{
    public class Medicine
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public string? ImageUrl { get; set; }
       /*[NotMapped]
        public IFormFile? ImageFile { get; set; }
        */public string? Company { get; set; }
        public string? MadeIn { get; set; }
        public decimal? Price { get; set; }
        public DateTime? DateCreated { get; set; }   // تاريخ الانشاء
        public DateTime? ExpirationDate { get; set; }    // تاريخ الانتهاء 

        public List<PharmacyMedicine> PharmacyMedicines { get; set; }

        public int? CategoryId { get; set; }
        public Category? Category { get; set; }
    }
}
